pkg/asset/agent/image/ignitionbase.go

```diff
@@ -0,0 +1,301 @@
+package image
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	igntypes "github.com/coreos/ignition/v2/config/v3_2/types"
+	"github.com/coreos/stream-metadata-go/arch"
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v2"
+
+	"github.com/openshift/assisted-service/models"
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/agent/manifests"
+	"github.com/openshift/installer/pkg/asset/agent/mirror"
+	"github.com/openshift/installer/pkg/asset/ignition"
+	"github.com/openshift/installer/pkg/asset/ignition/bootstrap"
+	"github.com/openshift/installer/pkg/rhcos"
+	"github.com/openshift/installer/pkg/types"
+	"github.com/openshift/installer/pkg/version"
+)
+
+const (
+	unconfiguredIgnitionFilename = "unconfigured-agent.ign"
+)
+
+// IgnitionBase is an asset that generates the agent installer base ignition filea
+// which excludes any cluster configuration files.
+type IgnitionBase struct {
+	Config                    *igntypes.Config
+	CPUArch                   string
+	File                      *asset.File
+	infraEnvID                string
+	archName                  string
+	osImage                   models.OsImage
+	hasMirrorConfig           bool
+	releaseImageMirror        string
+	publicContainerRegistries string
+}
+
+var agentEnabledServices = []string{
+	"agent-interactive-console.service",
+	"agent.service",
+	"assisted-service-db.service",
+	"assisted-service-pod.service",
+	"assisted-service.service",
+	"create-cluster-and-infraenv.service",
+	"node-zero.service",
+	"multipathd.service",
+	"selinux.service",
+	"install-status.service",
+	// Services disabled in IgnitionBase
+	// "set-hostname.service",
+	// "start-cluster-installation.service",
```

Comment:
I've found that if the dependencies are set such that that node-zero.service doesn't start until load-config.iso.service has run and created the `/etc/assisted/rendezvous-host.env` file, then these services shouldn't be removed as the are necessary and will start after the node-zero.service